<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>480</width>
    <height>390</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Tonemap - Imaginable</string>
  </property>
  <property name="windowIcon">
   <iconset resource="tonemap_gui.qrc">
    <normaloff>:/icon/tonemap</normaloff>:/icon/tonemap</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>4</number>
    </property>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <layout class="QGridLayout" name="gridLayout" rowstretch="1,0" columnstretch="1,0">
       <property name="margin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="1" column="0">
        <widget class="QScrollBar" name="horizontalScrollBar">
         <property name="maximum">
          <number>0</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QWidget" name="views" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1">
          <property name="spacing">
           <number>4</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="Preview" name="original_view">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>50</width>
              <height>50</height>
             </size>
            </property>
            <property name="text">
             <string>original</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Preview" name="preview">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>50</width>
              <height>50</height>
             </size>
            </property>
            <property name="text">
             <string>preview</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QScrollBar" name="verticalScrollBar">
         <property name="maximum">
          <number>0</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>480</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Open"/>
    <addaction name="action_Save"/>
    <addaction name="action_Save_as"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="action_Show_original"/>
    <addaction name="separator"/>
    <addaction name="action_Zoom_in"/>
    <addaction name="action_Zoom_out"/>
    <addaction name="action_Zoom_to_fit"/>
    <addaction name="action_Zoom_one"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_View"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
   <addaction name="separator"/>
   <addaction name="action_Zoom_in"/>
   <addaction name="action_Zoom_out"/>
   <addaction name="action_Zoom_to_fit"/>
   <addaction name="action_Zoom_one"/>
  </widget>
  <widget class="QStatusBar" name="status_bar"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Settings</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,1">
     <property name="spacing">
      <number>4</number>
     </property>
     <property name="margin">
      <number>4</number>
     </property>
     <item>
      <layout class="QGridLayout" name="_3">
       <property name="margin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>4</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_saturation">
         <property name="text">
          <string>&amp;Saturation gamma</string>
         </property>
         <property name="buddy">
          <cstring>slider_saturation</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="slider_saturation">
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBelow</enum>
         </property>
         <property name="tickInterval">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="value_saturation">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>+1.0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_blur">
         <property name="text">
          <string>Gaussian &amp;blur</string>
         </property>
         <property name="buddy">
          <cstring>slider_blur</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSlider" name="slider_blur">
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>20</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBelow</enum>
         </property>
         <property name="tickInterval">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSlider" name="slider_mix">
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>50</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBelow</enum>
         </property>
         <property name="tickInterval">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_mix">
         <property name="text">
          <string>&amp;Mix</string>
         </property>
         <property name="buddy">
          <cstring>slider_mix</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="value_mix">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>50%</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>4</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="value_blur_percent">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>40</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>20%</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="value_blur_pixels">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>0 px</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="show_original">
         <property name="text">
          <string>Show &amp;original</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="reset_settings">
         <property name="text">
          <string>&amp;Reset</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/file/open</normaloff>:/file/open</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/file/save</normaloff>:/file/save</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/file/quit</normaloff>:/file/quit</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="action_Zoom_in">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/zoom/in</normaloff>:/zoom/in</iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;in</string>
   </property>
  </action>
  <action name="action_Zoom_out">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/zoom/out</normaloff>:/zoom/out</iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;out</string>
   </property>
  </action>
  <action name="action_Show_original">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>S&amp;how original</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="action_Zoom_to_fit">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/zoom/fit</normaloff>:/zoom/fit</iconset>
   </property>
   <property name="text">
    <string>Zoom to fit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="action_Zoom_one">
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/zoom/one</normaloff>:/zoom/one</iconset>
   </property>
   <property name="text">
    <string>Zoom 100%</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="action_Save_as">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="tonemap_gui.qrc">
     <normaloff>:/file/saveas</normaloff>:/file/saveas</iconset>
   </property>
   <property name="text">
    <string>Save &amp;as</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Preview</class>
   <extends>QLabel</extends>
   <header>preview.hpp</header>
   <slots>
    <signal>shifted(int,int)</signal>
    <signal>resized(int,int)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="tonemap_gui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Open</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>313</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Save</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>313</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Save_as</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>233</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>313</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>show_original</sender>
   <signal>clicked(bool)</signal>
   <receiver>action_Show_original</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Show_original</sender>
   <signal>triggered(bool)</signal>
   <receiver>show_original</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>59</x>
     <y>346</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Show_original</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>showOriginal(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Zoom_in</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Zoom_out</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Zoom_one</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomOne()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Zoom_to_fit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomToFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setSaturation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel">
     <x>186</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_blur</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setBlur(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_mix</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setMix(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset_settings</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>resetSliders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>471</x>
     <y>356</y>
    </hint>
    <hint type="destinationlabel">
     <x>630</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalScrollBar</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>horizontallySlided(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>verticalScrollBar</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>verticallySlided(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>467</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>original_view</sender>
   <signal>shifted(int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>previewShifted(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preview</sender>
   <signal>shifted(int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>previewShifted(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>344</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preview</sender>
   <signal>resized(int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>previewResized(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>276</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>51</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>fileOpen()</slot>
  <slot>fileSave()</slot>
  <slot>fileSaveAs()</slot>
  <slot>showOriginal(bool)</slot>
  <slot>zoomIn()</slot>
  <slot>zoomOut()</slot>
  <slot>zoomOne()</slot>
  <slot>zoomToFit()</slot>
  <slot>setSaturation(int)</slot>
  <slot>setBlur(int)</slot>
  <slot>setMix(int)</slot>
  <slot>resetSliders()</slot>
  <slot>horizontallySlided(int)</slot>
  <slot>verticallySlided(int)</slot>
  <slot>previewResized(int,int)</slot>
  <slot>previewShifted(int,int)</slot>
 </slots>
</ui>
